### DISCUSSION TITLE
Export displacements as a vector variable.

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#21608] [Q&A General] [answered]
Hi, I'm trying to use VTK or Tecplot outputs to get around Exodus's inability to have elements of disparate types in the same block, however, when viewing a VTK or tecplot file, Paraview does not automatically detect and combine the three displacment variables into a vector to all the mesh to be displaced/deformed. I have attempted to solve this by making a LAGRANGE_VEC aux variable, but have been unable to find the correct aux kernel for combining a number of LAGRANGE variables into a LAGRANGE_VEC. BuildArrayVariableAux does not allow for the use of vectors, and if I use an array instead then the same issue persists where paraview reads the components as three separate variables instead of a vector.

**Comment by GiudGiud** [comment, marked as answer]
BuildArrayVariableAux is for array variables not vector variables.

You may just name your variables
var_x
var_y
var_z 
to get Paraview to recognize them as vectors. 

You could also add a BuildVectorVariableAux to the framework, we'd be happy to have this

Guillaume
